awswrangler/neptune/client.py

```diff
@@ -0,0 +1,205 @@
+        if region is not None:
+            return region
+        raise exceptions.InvalidArgument("There is no region_name defined on boto3, please configure it.")
+
+
+    def __ensure_session(self, session: boto3.Session = None) -> boto3.Session:
+        """Ensure that a valid boto3.Session will be returned."""
+        if session is not None:
+            return session
+        elif boto3.DEFAULT_SESSION:
+            return boto3.DEFAULT_SESSION
+        else:
+            return boto3.Session()
+
+
+    def _prepare_request(self, method, url, *, data=None, params=None, headers=None, service=NEPTUNE_SERVICE_NAME) -> requests.PreparedRequest:
+        request = requests.Request(method=method, url=url, data=data, params=params, headers=headers)
+        if self.boto3_session is not None:
+            aws_request = self._get_aws_request(method=method, url=url, data=data, params=params, headers=headers,
+                                                service=service)
+            request.headers = dict(aws_request.headers)
+
+        return request.prepare()
+
+
+    def _get_aws_request(self, method, url, *, data=None, params=None, headers=None, service=NEPTUNE_SERVICE_NAME) -> AWSRequest:
+        req = AWSRequest(method=method, url=url, data=data, params=params, headers=headers)
+        if self.iam_enabled:
+            credentials = self.boto3_session.get_credentials()
+            try:
+                frozen_creds = credentials.get_frozen_credentials()
+            except AttributeError:
+                print("Could not find valid IAM credentials in any the following locations:\n")
+                print("env, assume-role, assume-role-with-web-identity, sso, shared-credential-file, custom-process, "
+                      "config-file, ec2-credentials-file, boto-config, container-role, iam-role\n")
+                print("Go to https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html for more "
+                      "details on configuring your IAM credentials.")
+                return req
+            SigV4Auth(frozen_creds, service, self.region).add_auth(req)
+            prepared_iam_req = req.prepare()
+            return prepared_iam_req
+        else:
+            return req
+
+
+    def read_opencypher(self, query: str, headers:Dict[str, Any] = None) -> Dict[str, Any]:
+        if headers is None:
+            headers = {}
+
+        if 'content-type' not in headers:
+            headers['content-type'] = 'application/x-www-form-urlencoded'
+
+        url = f'{self._http_protocol}://{self.host}:{self.port}/openCypher'
+        data = {
+            'query': query
+        }
+
+        req = self._prepare_request('POST', url, data=data, headers=headers)
+        res = self._http_session.send(req)
+
+        return res.json()
+
+    
+    def read_gremlin(self, query, headers:Dict[str, Any] = None) -> Dict[str, Any]:
+        try:
+            nest_asyncio.apply()
```

Comment:
I'm curious - is this required? Nesting event loops seems to be a controversial topic

Reply:
I tried running it without nesting event loops and it does not work inside Jupyter without having it nested.

Reply:
So looks like they even [added](https://tinkerpop.apache.org/docs/3.5.1/upgrade/#_python_transport_layer) the ability to enable event loop nesting to the gremlin driver itself now. You can just enable it by passing `transport_factory=AiohttpTransport(call_from_event_loop=True)`. I added a commit with this